lib/mpool/lib/mblock_file.c

```diff
@@ -922,10 +958,17 @@ mblock_file_find(struct mblock_file *mbfp, uint64_t *mbidv, int mbidc, struct mb
         props->mpr_objid = *mbidv;
         props->mpr_mclass = mcid_to_mclass(mbfp->mcid);
         props->mpr_write_len = mblock_wlen_get(mbfp, *mbidv);
-        if (mblock_is_prealloc(mbfp, *mbidv))
+
+        if (mblock_is_punched(mbfp, *mbidv)) {
+            err2 = mblock_alen_get(mbfp, *mbidv, &props->mpr_alloc_cap);
+            if (!err2 && mblock_is_prealloced(mbfp, *mbidv))
+                props->mpr_alloc_cap += (mbfp->mblocksz - props->mpr_write_len);
+            assert(props->mpr_alloc_cap <= mbfp->mblocksz);
+        } else if (mblock_is_prealloced(mbfp, *mbidv)) {
             props->mpr_alloc_cap = mbfp->mblocksz;
-        else
+        } else {
             props->mpr_alloc_cap = props->mpr_write_len;
```

Comment:
Use a local var in all the branch code blocks and one assignment to `props->mpr_alloc_cap` at the end.